src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -30,26 +35,130 @@ object TrialService {
+          state = Some(Terminated),
+          terminatedDate = Instant.now
+        )
+
+    updateUserState(managerInfo, user, statusTransition)
+  }
+
+  private def updateUserState(managerInfo: UserInfo,
+                              user: String,
+                              statusTransition: UserTrialStatus => UserTrialStatus): Future[PerRequestMessage] = {
+    // TODO: Handle unregistered users which get 404 from Sam causing adminGetUserByEmail to throw
+    // TODO: Handle errors that may come up while querying Sam
+    for {
+      regInfo <- samDao.adminGetUserByEmail(RawlsUserEmail(user))
+      workbenchUserInfo = WorkbenchUserInfo(regInfo.userInfo.userSubjectId, regInfo.userInfo.userEmail)
+      result <- updateTrialStatus(managerInfo, workbenchUserInfo, statusTransition)
+    } yield result match {
+      case StatusUpdate.Success => RequestComplete(NoContent)
+      case StatusUpdate.Failure => RequestComplete(InternalServerError)
+    }
+  }
+
+  private def updateTrialStatus(managerInfo: UserInfo,
+                                userInfo: WorkbenchUserInfo,
+                                statusTransition: UserTrialStatus => UserTrialStatus): Future[StatusUpdate.Attempt] = {
+    // Create hybrid UserInfo with the managerInfo's credentials and users' subjectIds
+    val sudoUserInfo = managerInfo.copy(id = userInfo.userSubjectId)
+
+    // Use the hybrid UserInfo while querying Thurloe
+    thurloeDao.getTrialStatus(sudoUserInfo) map {
+      case Some(currentStatus) =>
+        val currentState = currentStatus.state
+        val newStatus = statusTransition(currentStatus)
+        val newState = newStatus.state
+
+        if (currentState == newState) {
+          logger.warn(
+            s"The user '${userInfo.userEmail}' is already in the trial state of '$newState'. " +
+            s"No further action will be taken.")
+
+          StatusUpdate.Success
+        } else {
+          logger.warn(s"Current trial state of the user '${userInfo.userEmail}' is '$currentState'")
+          logger.warn("Checking if enabling is allowed from that state...")
+
+          require(newState.nonEmpty, "Cannot transition to an unspecified state")
+
+          // TODO Handle invalid initial trial status by
+          //    -adding another case to Trial.StatusUpdate
+          //    -using that case to return a BadRequest from the parent method (i.e. updateUserState())
+          assert(newState.get.isAllowedFrom(currentState),
+            s"Cannot transition from $currentState to $newState")
+
+          // Save updates to user's trial status
+          thurloeDao.saveTrialStatus(sudoUserInfo, newStatus)
+          logger.warn(s"Updated profile saved as $newState; we are done!")
+
+          StatusUpdate.Success
+        }
+      // TODO: Handle case where TrialStatusOpt is None, i.e., if the user profile doesn't exist
+      // in Thurloe, create one and set to the new state along with the other tags
```

Comment:
I don't think we should automatically create keys in Thurloe in this instance. Given the leniency of the conversion from `ProfileWrapper` to `UserTrialStatus`, I believe you should only get a `None` here if something went wrong. Also, based on the business logic, the only time we should be creating a trial status is during an enable operation; not disable, terminate, or enroll.